cmd/smartcrop/main.go

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2014-2019 Christian Muehlhaeuser
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *	Authors:
+ *		Christian Muehlhaeuser <muesli@gmail.com>
+ *		Michael Wendland <michael@michiwend.com>
+ *		Bjørn Erik Pedersen <bjorn.erik.pedersen@gmail.com>
+ */
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"image"
+	"image/jpeg"
+	"image/png"
+	"io"
+	"os"
+
+	"github.com/muesli/smartcrop"
+	"github.com/muesli/smartcrop/nfnt"
+)
+
+func main() {
+	input := flag.String("input", "", "input filename")
+	output := flag.String("output", "", "output filename")
+	w := flag.Int("width", 0, "crop width")
+	h := flag.Int("height", 0, "crop height")
+	resize := flag.Bool("resize", true, "resize after cropping")
+	quality := flag.Int("quality", 85, "jpeg quality")
+	flag.Parse()
+
+	if *input == "" {
+		fmt.Fprintln(os.Stderr, "No input file given")
+		return
+	}
+
+	f, err := os.Open(*input)
+	defer f.Close()
```

Comment:
Move this line below the error check. If an error occurred we must not call `f.Close()`.